projects/observability/src/shared/components/timeline-card-list/timeline-card-list.component.ts

```diff
@@ -30,4 +55,40 @@ export class TimelineCardListComponent {
   public readonly dateFormat: DateFormatOptions = {
     mode: DateFormatMode.TimeWithSeconds
   };
+
+  public items: TimelineListItem[] = [];
+  public allCards: TimelineCardContainerComponent[] = [];
+
+  public ngAfterContentInit(): void {
+    queryListAndChanges$(this.cards)
+      .pipe(map(list => list.toArray()))
+      .subscribe(allCards => {
+        this.allCards = allCards;
+        this.buildItems(allCards);
+      });
+  }
+
+  public readonly isSelectedCard = (card: TimelineCardContainerComponent): boolean =>
+    this.selectedIndex !== undefined && this.selectedIndex >= 0 && this.allCards[this.selectedIndex] === card;
+
+  private buildItems(allCards: TimelineCardContainerComponent[]): void {
```

Comment:
What if we changed this to create groups rather than a boolean referring to a sibling. Like:
```html
    <ht-timeline-card-list>
      <ht-timeline-card-group [collapsed]="true"> <!-- could add things like summary, etc in the future to this -->
        <ht-timeline-card-container [timestamp]="cardData.timestamp">
          <div class="custom-card">
            <div class="title">{{cardData.name}}</div>
          </div>
        </ht-timeline-card-container>
      </ht-timeline-card-group>
    </ht-timeline-card-list>
```

That API seems clearer to me, but I'm not sure on how we're using it.

Reply:
The reason I didn't chose this is that it allows random grouping of cards. The grouping we want has to be strictly adjacent in time. Otherwise, **we are hiding important information from the user** and the timeline view will also be broken.

> This API strikes me as pretty strange - components don't generally know about their siblings.

Yes, it is strange. But it is also on a container component and not on a generalized component. The `<ht-timeline-card-list>` is responsible for rendering the cards, the containers are just supposed to provide meta info for the cards. With this understanding, i think it makes sense. And, if `ht-timeline-card-container` is a structural directive then the same would look much more decent :)

Reply:
So what are the criteria we'd generally group by? Timestamps are within some threshold of each other and card data is also matching some implementation-specific equality (e.g. same title) ?

I still like the idea of telling the timeline list how to group with some form of metadata, vs a boolean back reference which forces logic on either side. For example, passing a grouping function, or a grouping key + time threshold.

Reply:
So exactly preceding in time and same type. 

The grouping function approach still has the potential to break the timeline imo. I want to avoid a scenario where we are showing cards out of order of timestamps. With a grouping function that can happen if the function itself is written incorrectly. And we will also have to maintain unique grouping keys for correctness.

In the current implementation, the adjacent grouping is strictly enforced. If EventA occurs again after EventB, then they are treated differently. Current api is ensuring this criteria.

Reply:
I guess I'd think that functiin would be used only to check boundaries, while the timie is stsill enforced. Anywayy, I'm ok with htis prroceediinig. can revissitt if needed. (sorry mac keyboard)